models/imagick.go

```diff
@@ -49,6 +55,10 @@ func (p *IMagick) Process(w http.ResponseWriter, r *http.Request, args *ProcessA
 		}
 	}
 
+	// cache that image as rending is costly and url guaranties uniqueness
+	// as long a original source has content based hashes for it's assets
+	// TODO(kiasaki) Be nice guys and do the cache based on source ourselves?
+	w.Header().Set("Cache-control", "public, max-age=259200")
```

Comment:
Cloudfront just since I've used it before. Was pretty easy to get going with.

```
› dig *.firesize.com

; <<>> DiG 9.8.3-P1 <<>> *.firesize.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 47915
;; flags: qr rd ra; QUERY: 1, ANSWER: 9, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;*.firesize.com.                        IN      A

;; ANSWER SECTION:
*.firesize.com.         3584    IN      CNAME   dh41b7aq2g8bl.cloudfront.net.
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.207.82
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.205.92
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.206.214
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.207.102
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.207.116
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.207.107
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.192.206.203
dh41b7aq2g8bl.cloudfront.net. 44 IN     A       54.230.207.145

;; Query time: 2 msec
;; SERVER: 10.0.1.1#53(10.0.1.1)
;; WHEN: Thu Mar 26 23:00:51 2015
;; MSG SIZE  rcvd: 202
```